### PR TITLE
feat: convert WorkoutEntry to a record class

### PR DESCRIPTION
**Pull request by frank-connolly**
Records are just better for this use case

### CODE REVIEW

File: src/main/java/com/frank/ironbank/models/WorkoutEntry.java [hunk 1]
```diff
@@ -1,67 +1,12 @@
 package com.frank.ironbank.models;
 
-import jakarta.persistence.*;
```
Comment (tmccroskery) [inline src/main/java/com/frank/ironbank/models/WorkoutEntry.java:3]: mad packages man :O

Never heard of em

File: src/main/java/com/frank/ironbank/models/WorkoutEntry.java [hunk 2]
```diff
@@ -1,67 +1,12 @@
...
-
-    public Long getId() {
-        return id;
-    }
-
-    public LocalDateTime getStartTime() {
-        return startTime;
-    }
-
-    public void setStartTime(LocalDateTime startTime) {
```
Comment (tmccroskery) [inline src/main/java/com/frank/ironbank/models/WorkoutEntry.java:32]: should there be checks on the set values? 

like if startTime == NULL or some mad jazz like that?

File: src/main/java/com/frank/ironbank/models/WorkoutEntry.java [hunk 3]
```diff
@@ -1,67 +1,12 @@
...
-
-    public String getNotes() {
-        return notes;
-    }
-
-    public void setNotes(String notes) {
-        this.notes = notes;
-    }
-}
+public record WorkoutEntry(
```
Comment (tmccroskery) [inline src/main/java/com/frank/ironbank/models/WorkoutEntry.java:5]: oh fuck wait I assume this automatically provides get/set for each element?

File: src/main/java/com/frank/ironbank/models/WorkoutEntry.java (hunk 1 above)
Comment (frank-connolly) [reply src/main/java/com/frank/ironbank/models/WorkoutEntry.java:3]: Its just some Java nonsense. Jakarta is still core Java its not anything crazy. 
https://www.youtube.com/watch?v=0l0ZtoWgOB0&ab_channel=in28minutes-GetCloudCertified

File: src/main/java/com/frank/ironbank/models/WorkoutEntry.java (hunk 3 above)
Comment (frank-connolly) [reply src/main/java/com/frank/ironbank/models/WorkoutEntry.java:5]: It gives us getters but no setters. You can't change the record - you gotta make a new one. This is a good explainer:
https://www.youtube.com/watch?v=gJ9DYC-jswo&ab_channel=CodingwithJohn

File: src/main/java/com/frank/ironbank/models/WorkoutEntry.java (hunk 2 above)
Comment (frank-connolly) [reply src/main/java/com/frank/ironbank/models/WorkoutEntry.java:32]: Aye the record approach fixes this cos the whole object is immutable. No setters

**Comment by frank-connolly** [comment]
I'm gonna merge this

**Pull request merged**